File: app/src/main/java/com/pirateman/exercisemanager/selectedexercise/SelectedExerciseRepository.java
package com.pirateman.exercisemanager.selectedexercise;

import android.app.Application;
import android.arch.lifecycle.LiveData;

import com.pirateman.exercisemanager.exercise.Exercise;
import com.pirateman.exercisemanager.exercise.ExerciseActivity;
import com.pirateman.exercisemanager.exercise.ExerciseDao;
import com.pirateman.exercisemanager.exercise.ExerciseDatabase;

import java.util.List;
import java.util.concurrent.Executor;
import java.util.concurrent.Executors;

public class SelectedExerciseRepository
{
    private ExerciseDao exerciseDao;
    private LiveData<List<Exercise>> exercises;

    SelectedExerciseRepository(Application application)
    {
        ExerciseDatabase exerciseDatabase = ExerciseDatabase.getINSTANCE(application);
        exerciseDao = exerciseDatabase.exerciseDao();

        exercises = exerciseDao.getAll();
    }

    public LiveData<List<Exercise>> getAllExercises()
    {
        return exercises;
    }

    public void insertExercise(final Exercise exercise)
    {
        Executor executor = Executors.newSingleThreadExecutor();
        executor.execute(() -> exerciseDao.insertExercise(exercise));

    }
}

File: app/src/main/java/com/pirateman/exercisemanager/selectedexercise/SelectedExerciseRepository.kt
package com.pirateman.exercisemanager.selectedexercise

import android.app.Application
import androidx.lifecycle.LiveData
import com.pirateman.exercisemanager.exercise.Exercise
import com.pirateman.exercisemanager.exercise.ExerciseDao
import com.pirateman.exercisemanager.exercise.ExerciseDatabase
import java.util.concurrent.Executor
import java.util.concurrent.Executors

class SelectedExerciseRepository internal constructor(application: Application?) {
    //private val exerciseDao: ExerciseDao?
    //val allExercises: LiveData<List<Exercise?>?>?

    fun insertExercise(exercise: Exercise?) {
//        val executor: Executor = Executors.newSingleThreadExecutor()
//        executor.execute { exerciseDao?.insertExercise(exercise) }
    }

    init {
       // val exerciseDatabase = ExerciseDatabase.getINSTANCE(application)
        //exerciseDao = exerciseDatabase?.exerciseDao()
        //allExercises = exerciseDao?.all
    }
}